cmd/kubeadm/app/cmd/phases/preflight.go

```diff
@@ -56,6 +55,52 @@ var (
 	errorMissingConfigFlag = errors.New("the --config flag is mandatory")
 )
 
+// preflightMasterContext defines the behavior that a runtime context passed to the PreflightMaster master phase
+// should have. Please note that we are using an interface in order to make this phase reusable in different workflows
+// (and thus with different runtime context, all of them requested to be compliant to this interface)
+type preflightMasterContext interface {
+	Cfg() *kubeadmapi.InitConfiguration
+	DryRun() bool
+	IgnorePreflightErrors() sets.String
+}
+
+// PreflightMaster defines that kubeadm workflow phase that implements preflight checks for a new master node.
+var PreflightMaster = workflow.SimplePhase{
+	Name:    "preflight",
+	Short:   "Run master pre-flight checks",
+	Long:    "Run master pre-flight checks, functionally equivalent to what implemented by kubeadm init.",
+	Example: masterPreflightExample,
+	Run:     runPreflightMaster,
+}
+
+// runPreflightMaster executes preflight checks logic.
+func runPreflightMaster(c workflow.RunnerContext) error {
+	ctx, ok := c.(preflightMasterContext)
+
+	if !ok {
+		return fmt.Errorf("preflight phase invoked with an invalid context")
+	}
+
+	fmt.Println("[preflight] running pre-flight checks")
+	if err := preflight.RunInitMasterChecks(utilsexec.New(), ctx.Cfg(), ctx.IgnorePreflightErrors()); err != nil {
+		return nil
+	}
+
+	// TODO: consider if to split this into a nested phase preflight/pull
```

Comment:
yes, we should probably consider something like that.

e.g.
...
preflight/validate
preflight/image-pull
...